core/new-gui/src/app/workspace/component/result-panel/result-panel.component.ts

```diff
@@ -224,18 +263,72 @@ export class ResultPanelComponent {
     this.breakpointAction = false;
   }
 
+  /**
+   * Callback function for table query params changed event
+   *   params containing new page index, new page size, and more
+   *   (this function will be called when user switch page)
+   *
+   * @param params new parameters
+   */
+  public onTableQueryParamsChange(params: NzTableQueryParams) {
+    const { pageSize: newPageSize, pageIndex: newPageIndex } = params;
+    this.currentPageSize = newPageSize;
+    this.currentPageIndex = newPageIndex;
+
+    if (this.isFrontPagination) {
+      return;
+    }
+
+    this.isLoadingResult = true;
+    this.workflowWebsocketService.send('ResultPaginationRequest', { pageSize: newPageSize, pageIndex: newPageIndex });
+    this.workflowWebsocketService.websocketEvent().subscribe(websocketEvent => {
+      if (websocketEvent.type !== 'PaginatedResultEvent') {
+        return;
+      }
+
+      const highlightedOperators = this.workflowActionService.getJointGraphWrapper().getCurrentHighlightedOperatorIDs();
+
+      for (const result of websocketEvent.paginatedResults) {
+        if (result.operatorID === highlightedOperators[0]) {
+          this.total = result.totalRowCount;
+          this.currentResult = result.table.slice();
+          this.isLoadingResult = false;
+          return;
+        }
+      }
+    });
+  }
+
   /**
    * Updates all the result table properties based on the execution result,
    *  displays a new data table with a new paginator on the result panel.
    *
-   * @param response
+   * @param resultData rows of the result (may not be all rows if displaying result for workflow completed event)
+   * @param totalRowCount if present, is the total number of rows for the result; otherwise, use length of resultData
    */
-  private setupResultTable(resultData: ReadonlyArray<object>) {
+  private setupResultTable(resultData: ReadonlyArray<object>, totalRowCount?: number) {
```

Comment:
I see the typescript interface that the totalRowCount is required to be not undefined. Any reason that we make it optional here? is there anywhere else that also calls this function without totalRowCount?

Reply:
My original thought is that this method is used for `ExecutionState.Paused`, `ExecutionState.Completed`, and `ExecutionState.BreakpointTriggered`. But only `ExecutionState.Completed` has pagination. Thus, I made this `totalRowCount` optional.

I can make it required, and modify the `ExecutionState.Paused` and `ExecutionState.BreakpointTriggered` parts in `displayResultPanel` to pass the row count into the method as well.

Reply:
Updated in commit https://github.com/Texera/texera/pull/903/commits/efb7d52062225eeab5dde1061f8d53ca13de412d